pkg/agent/memberlist/server.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memberlist
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/memberlist"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+)
+
+const defaultInterval = 30 * time.Second
+
+type Server struct {
+	bindPort        int
+	nodeConfig      *config.NodeConfig
+	nodeInformer    coreinformers.NodeInformer
+	memberList      *memberlist.Memberlist
+	memberRWLock    sync.RWMutex
+	existingMembers *[]string
+	defaultInterval time.Duration
+}
+
+func NewMemberlistServer(p int, nodeInformer coreinformers.NodeInformer, nodeConfig *config.NodeConfig) (*Server, error) {
+	klog.Infof("Node config: %#v", nodeConfig)
+
+	s := &Server{
+		bindPort:        p,
+		nodeInformer:    nodeInformer,
+		nodeConfig:      nodeConfig,
+		defaultInterval: defaultInterval,
+	}
+
+	hostname := s.nodeConfig.Name
+	bindPort := s.bindPort
+	hostIP := s.nodeConfig.NodeIPAddr.IP
+
+	nodeMember := fmt.Sprintf("%s:%d", hostIP.String(), bindPort)
+
+	klog.Infof("Add new node: %s", nodeMember)
+
+	conf := memberlist.DefaultLocalConfig()
+	conf.Name = hostname + "-" + strconv.Itoa(bindPort)
+
+	conf.BindPort = bindPort
+	conf.AdvertisePort = bindPort
+
+	klog.Infof("Configs: %+v\n", conf)
```

Comment:
klog already appends newline.
And it seems that all logs are in default level?

Reply:
Removed ‘\n’， mistake.  Do we have a specification about log level?